src/nvim/path.c

```diff
@@ -2186,9 +2186,16 @@ static int path_get_absolute_path(const char_u *fname, char_u *buf,
 
 /// Check if the given file is absolute.
 ///
-/// This just checks if the file name starts with '/' or '~'.
 /// @return `TRUE` if "fname" is absolute.
 int path_is_absolute_path(const char_u *fname)
 {
+#ifdef WIN32
+  // A name like "d:/foo" and "//server/share" is absolute
+  return (isalpha(fname[0]) && fname[1] == ':'
+          && vim_ispathsep_nocolon(fname[2]))
+           || (fname[0] == fname[1] && vim_ispathsep_nocolon(fname[0]));
```

Comment:
If `fname` can be an empty string, the check `fname[0] == fname[1]` should not be used (here), because `fname[1]` is not part of the string. The arguments of `&&` could be switched to fix this.